airflow/providers/amazon/aws/operators/datasync.py

```diff
@@ -27,25 +25,45 @@
 from airflow.utils.decorators import apply_defaults
 
 
-class AWSDataSyncCreateTaskOperator(BaseOperator):
-    r"""Create an AWS DataSync Task.
+# pylint: disable=too-many-instance-attributes, too-many-arguments
+class AWSDataSyncOperator(BaseOperator):
+    r"""Find, Create, Update, Execute and Delete AWS DataSync Tasks.
+
+    If ``do_xcom_push`` is True, then the TaskArn and TaskExecutionArn which
+    were executed will be pushed to an XCom.
 
-    If there are existing Locations which match the specified
-    source and destination URIs then these will be used for the Task.
-    Otherwise, new Locations can be created automatically,
-    depending on input parameters.
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AWSDataSyncOperator`
 
-    If ``do_xcom_push`` is True, the TaskArn which is created
-    will be pushed to an XCom.
+    .. note:: There may be 0, 1, or many existing DataSync Tasks. The default
+        behavior is to create a new Task if there are 0, or execute the Task
+        if there was 1 Task, or fail if there were many Tasks.
 
     :param str aws_conn_id: AWS connection to use.
-    :param str source_location_uri: Source location URI.
+    :param int wait_for_task_execution: Time to wait between two
```

Comment:
Name of that parameter suggest it's a boolean and that it determines whether to wait for task execution. Better name would be `task_execution_wait_time" I think.

Reply:
Typo, thanks for catching it. The code called this "wait_interval_seconds", so I've used that now. Checked the rest and I noticed that delete_task_after_execution wasn't documented so I fixed that now :)